consul/agent.go

```diff
@@ -99,8 +105,38 @@ func (a *Agent) Deregister(services []ServiceInstance) error {
 }
 
 // NewAgent returns a new Agent.
-func NewAgent() *Agent {
-	consulClient, _ := api.NewClient(api.DefaultConfig())
+func NewAgent(tokenFile string) *Agent {
+	config := api.DefaultConfig()
+	// at this point Token is empty string, if token was not passed on commandline we will just use not secure client
+	config.Token = getAgentToken(tokenFile)
+	config.Address = getAgentAddress()
+	consulClient, _ := api.NewClient(config)
 	agent := consulClient.Agent()
 	return &Agent{agentClient: agent}
 }
+
+func getAgentToken(tokenFile string) string {
```

Comment:
How about using built-in consul configuration https://github.com/hashicorp/consul/blob/v1.0.6/api/api.go#L29-L31

Reply:
Changed to standard consul environment var, but token from file should have precedence over env.